app/actions/websocket/teams.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DeviceEventEmitter} from 'react-native';
+
+import {localRemoveUserFromTeam} from '@actions/local/team';
+import {updateUsersNoLongerVisible} from '@actions/remote/user';
+import {queryCurrentTeamId} from '@app/queries/servers/system';
+import {queryCurrentUser} from '@app/queries/servers/user';
+import {isGuest} from '@app/utils/user';
+import DatabaseManager from '@database/manager';
+
+export async function handleLeaveTeamEvent(serverUrl: string, msg: any) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const currentTeamId = await queryCurrentTeamId(database.database);
+    const user = await queryCurrentUser(database.database);
+    if (!user) {
+        return;
+    }
+
+    if (user.id === msg.data.user_id) {
+        localRemoveUserFromTeam(serverUrl, msg.data.team_id);
+
+        // TODO originally it also removed channel categories and threads in teams. Apparently none of those features are yet implemented.
+        // TODO consider whether localRemove should also remove team channel history and team search history.
+        if (isGuest(user.roles)) {
+            updateUsersNoLongerVisible(serverUrl);
+        }
+        if (currentTeamId === msg.data.team_id) {
+            DeviceEventEmitter.emit('leave_team');
```

Comment:
I think that instead of emitting an event to be handled elsewhere that could cause unexpected behavior as the current team Id won't have a record in the db, we should move this before deleting the current team then select the next team here, send the user to the home screen or to the no teams screen (this one still pending), set the current channel as an empty string all with `setCurrentTeamAndChannelId` and then emit the event just to show an Alert box notifying the user if needed.

Reply:
Mentions will be cross team, searches may become cross team, and account is team independent. Therefore... we should only move the user if they are in post screens, right?

Reply:
Home screen contains the four tabs but is still the home screen 😀